dbms/src/DataStreams/HashJoinProbeBlockInputStream.cpp

```diff
@@ -38,16 +37,29 @@ Block HashJoinProbeBlockInputStream::getTotals()
     if (auto * child = dynamic_cast<IProfilingBlockInputStream *>(&*children.back()))
     {
         totals = child->getTotals();
-        join_probe_actions->executeOnTotals(totals);
+        if (!totals)
+        {
+            if (join && join->hasTotals())
+            {
+                for (const auto & name_and_type : getHeader().getColumnsWithTypeAndName())
```

Comment:
Although `getTotals` might be useless for us, but I think using `getHeader()` here is not right, because the `totals` should only contain information about `child`(since `totals = child->getTotals()`), but `getHeader()` contains information about `join` itself.

Reply:
> Although `getTotals` might be useless for us, but I think using `getHeader()` here is not right, because the `totals` should only contain information about `child`(since `totals = child->getTotals()`), but `getHeader()` contains information about `join` itself.

done.